pkg/preflight/preflight_utils.go

```diff
@@ -377,3 +388,116 @@ func executeRemoteCommand(clientset *kubernetes.Clientset, config *rest.Config,
 	err := execute("POST", req.URL(), config, nil, &stdout, &stderr, tty)
 	return strings.TrimSpace(stdout.String()), strings.TrimSpace(stderr.String()), err
 }
+
+func waitForDeployment(clientset *kubernetes.Clientset, namespace string, pfDeployment *appsv1.Deployment) error {
+	timeout := time.NewTicker(2 * time.Minute)
+	defer timeout.Stop()
+	var d *appsv1.Deployment
+	var err error
+WAIT:
+	for {
+		d, err = getDeployment(clientset, namespace, pfDeployment.GetName())
+		if err != nil {
+			err = fmt.Errorf("error: unable to retrieve deployment: %s\n", pfDeployment.GetName())
+			fmt.Println(err)
+			return err
+		}
+		select {
+		case <-timeout.C:
+			break WAIT
+		default:
+			if int(d.Status.ReadyReplicas) > 0 {
+				break WAIT
+			}
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
It's a hard loop... With label breaks... It's not a huge deal, but is there any way we could make it a bit easier? Also, am I correct in my understanding that even on detecting success we still block here for 5 seconds?

Reply:
As far as I know, when the condition is met it will break out of the loop. The sleep is for the next iteration, when there the condition is not met or timeout happens.

Reply:
Thanks @ashwathishiva, I get it now...

Reply:
I just noticed that we have a few more of almost the exact copies of this. Do we have tests for these? I assume testing these would be pretty hard too. Any way we can generalize this into something like `waitFor()` and pass the function pointer with a check to make, other function pointers with the success/failure check. Then we could write an automated test for this functionality without any dependencies on k8s apis.

Reply:
Will get to writing tests for these soon. And yes, will generalize the wait functions too.